### PR TITLE
Construct exec and ctx only once in cond op to speed up

### PR DESCRIPTION
**Pull request by zh794390558**
### PR types

Performance optimization 
### PR changes

Others
### Describe

more refs: https://github.com/PaddlePaddle/Paddle/pull/45794

U2/U2++ model FP32 RTF  0.3787 -> 0.3425， 相对提升 9.6%

### CODE REVIEW

File: paddle/fluid/operators/controlflow/conditional_block_infer_op.cc [hunk 1]
```diff
@@ -73,14 +78,29 @@ class ConditionalBlockInferOp : public ConditionalOp {
...
+#endif
+      }
+      exec->RunPreparedContext(ctx.get(), &cur_scope, false, true, false);
       scope.DeleteScope(scopes->front());
     }
   }
+
+ private:
+  mutable std::shared_ptr<framework::Executor> exec{nullptr};
+  mutable std::unique_ptr<framework::ExecutorPrepareContext> ctx{nullptr};
```
Comment (Aurelius84) [inline paddle/fluid/operators/controlflow/conditional_block_infer_op.cc:103]: 按照规范，类的成员变量需要以下划线结尾，可以单独提PR修改，或再提个commit fix下

**Review by Aurelius84 — APPROVED** [review]
LGTM for the reset

File: paddle/fluid/operators/controlflow/conditional_block_infer_op.cc [hunk 2]
```diff
@@ -73,14 +78,29 @@ class ConditionalBlockInferOp : public ConditionalOp {
...
-      exec.Run(*block->Program(), &cur_scope, block->ID(), false);
+
+      if (!exec || !platform::is_same_place(exec->GetPlace(), dev_place)) {
+        auto &pdesc = *block->Program();
+        exec.reset(new framework::Executor(dev_place));
+        if (FLAGS_use_mkldnn) exec->EnableMKLDNN(pdesc);
+        ctx = exec->Prepare(
+            pdesc, block->ID(), std::vector<std::string>(), false);
+#ifdef PADDLE_WITH_MKLDNN
+        platform::AttachPointerHashToMKLDNNKey(exec.get(), dev_place);
```
Comment (yeliang2258) [inline paddle/fluid/operators/controlflow/conditional_block_infer_op.cc:92]: 放在if (FLAGS_use_mkldnn)判断生效之后

File: paddle/fluid/operators/controlflow/conditional_block_infer_op.cc [hunk 3]
```diff
@@ -73,14 +78,29 @@ class ConditionalBlockInferOp : public ConditionalOp {
...
-      framework::Executor exec(dev_place);
       auto *block = Attr<framework::BlockDesc *>("sub_block");
       VLOG(3) << "Conditional block.idx = " << block->ID()
               << ", scope = " << &cur_scope;
-      exec.Run(*block->Program(), &cur_scope, block->ID(), false);
+
+      if (!exec || !platform::is_same_place(exec->GetPlace(), dev_place)) {
+        auto &pdesc = *block->Program();
+        exec.reset(new framework::Executor(dev_place));
+        if (FLAGS_use_mkldnn) exec->EnableMKLDNN(pdesc);
```
Comment (yeliang2258) [inline paddle/fluid/operators/controlflow/conditional_block_infer_op.cc:88]: #ifdef PADDLE_WITH_MKLDNN
#endif
之间

**Pull request merged**